### PR TITLE
Compatibility update for Keras 2.1.3

### PR DESCRIPTION
**Pull request by catherio**
In Keras 2.1.3, inbound_nodes has been renamed to _inbound_nodes. Use a hasattr check to choose which property to use, and warn the user that support for the <2.1.3 behavior may be dropped.

see https://github.com/keras-team/keras/commit/958239c621a6be037c5f8b30be9270310735f725#diff-fe3ca734e8421a17729d0b0ea4c800ba

### CODE REVIEW

**Comment by catherio** [comment]
addresses https://github.com/tensorflow/cleverhans/issues/340

File: cleverhans/utils_keras.py [hunk 1]
```diff
@@ -138,7 +139,16 @@ def _get_logits_name(self):
         """
         softmax_name = self._get_softmax_name()
         softmax_layer = self.model.get_layer(softmax_name)
-        node = softmax_layer.inbound_nodes[0]
+
+        if hasattr(softmax_layer, 'inbound_nodes'):
+          warnings.warn(
+              "Please update your version to keras >= 2.1.3; "
+              "future verions of CleverHans may drop support for this version"
```
Comment (npapernot) [inline cleverhans/utils_keras.py]: Could you add a 6 month deprecation warning instead? They are easier to keep track of

**Comment by npapernot** [comment]
I left a minor comment and PEP8 is failing, but once that is fixed LGTM

```
/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py:17:1: E402 module level import not at top of file
/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py:19:1: E302 expected 2 blank lines, found 1
/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py:144:11: E111 indentation is not a multiple of four
/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py:148:11: E111 indentation is not a multiple of four
/home/travis/build/tensorflow/cleverhans/cleverhans/utils_keras.py:150:11: E111 indentation is not a multiple of four
```

**Comment by catherio** [comment]
The build won't be happy unless I fix both 339 and 340 at once, so I'm closing this and submitting one PR for both.

**Pull request closed without merging**